app/src/main/java/com/base/firebase_operations/MainActivity2.kt

```diff
@@ -63,4 +70,18 @@ class MainActivity2 : AppCompatActivity() {
             }
         }
     }
+    private fun DownloadImage(filename:String) = CoroutineScope(Dispatchers.IO).launch {
```

Comment:
First thing is that we have created a function that would be downloading the image from the storage and here we are setting a maxsize for the downloaded content so the reason for doing it is that we ensure that large files are not downloaded unnecessarily when the user have the issue with internet or limited data plan. So files which exceed the max  download size would not be downloaded because of the above reason and some performance reason for our application.
Now we then use the getBytes for getting the bytes of the image and it takes the maxsizedownload parameter ensuring the max size of the image to be downloaded.
Now we have the bytes of the image but we need it to be displayed in an UI component so here we are converting the bytes to an bitmap using the Bitmap Factory then just switching the context and displaying that into the image view.